html/rendering/non-replaced-elements/the-hr-element-0/setting-overflow-visible.html

```diff
@@ -18,27 +27,38 @@
 }
 </style>
 
-<div class=float></div>
-<hr id=test-control>
-<div class=clear></div>
+<div class=wrapper>
+ <div class=float></div>
+ <hr id=test-control>
+ <div class=float></div>
+</div>
 
-<div class=float></div>
-<hr id=test-visible>
-<div class=clear></div>
+<div class=wrapper>
+ <div class=float></div>
+ <hr id=test-visible>
+ <div class=float></div>
+</div>
 
 <script>
-const defaultBodyMargin = 8;
 
 test(() => {
   const hr = document.getElementById('test-control');
-  assert_equals(hr.offsetLeft, 50 + defaultBodyMargin, 'offsetLeft');
-  assert_equals(hr.clientHeight, 50, 'clientHeight');
+  assert_equals(hr.offsetLeft, 50, 'hr.offsetLeft');
+  assert_equals(hr.offsetTop, 0, 'hr.offsetTop');
+  assert_equals(hr.clientHeight, 50, 'hr.clientHeight');
+  const divAfter = hr.nextElementSibling;
+  assert_equals(divAfter.offsetLeft, 0, 'divAfter.offsetLeft');
+  assert_equals(divAfter.offsetTop, 50 + 1 + 1 /* hr border */, 'divAfter.offsetTop');
 }, 'control');
 
 test(() => {
   const hr = document.getElementById('test-visible');
-  assert_equals(hr.offsetLeft, 0 + defaultBodyMargin, 'offsetLeft');
-  assert_equals(hr.clientHeight, 0, 'clientHeight');
+  assert_equals(hr.offsetLeft, 0, 'hr.offsetLeft');
+  assert_equals(hr.offsetTop, 0, 'hr.offsetTop');
+  assert_equals(hr.clientHeight, 0, 'hr.clientHeight');
+  const divAfter = hr.nextElementSibling;
+  assert_equals(divAfter.offsetLeft, 50 + 50, 'divAfter.offsetLeft');
+  assert_equals(divAfter.offsetTop, 1 + 1 /* hr border */, 'divAfter.offsetTop');
```

Comment:
https://www.w3.org/TR/CSS22/visuren.html#float-rules

> A floating box's outer top may not be higher than the top of its containing block. When the float occurs between two collapsing margins, the float is positioned as if it had an otherwise empty anonymous block parent taking part in the flow. The position of such a parent is defined by the rules in the section on margin collapsing.

It's not super-clear to me, but maybe such an anonymous block parent is supposed to be there even if the float isn't between two collapsing margins?